### DISCUSSION TITLE
Groq support

### DISCUSSION THREAD

**Discussion by nganju98** [betalgo/openai#587] [Ideas]
Groq has claimed to be compatible with OpenAI.  With the OpenAI python or javascript librarires you can simply change the endpoint to a Groq url and it will just work.  See here: https://console.groq.com/docs/openai

If I create a pull request adding Groq support for chat completions will you merge it?  Or is there some objection to adding Groq support?

Thanks!

**Comment by kayhantolga** [comment]
We also support changing base url => https://github.com/betalgo/openai/discussions/572

I will not accept any PR that includes support for any competitors for a couple of reasons.

1. This library is promoted in OpenAI documentation as a Community Library. I don't think it would be a good idea to add support for competitors.
2. I am planning to create an SDK that supports all leading models. I understand most of the libraries use a similar API set with OpenAI, but I don’t think this approach will continue for a long time. A new library that can cover these differences would be a better idea, I think.

Also, it would be nice to get feedback about these, so please share your feedbacks.
